src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -5096,6 +5119,87 @@ void  MacroAssembler::decode_heap_oop_not_null(Register dst, Register src) {
   }
 }
 
+static Register pick_different_tmp(Register dst, Register src) {
+  Register tmp = r0;
+  if (tmp == src || tmp == dst) {
+    tmp = r1;
+    if (tmp == src || tmp == dst) {
+      tmp = r2;
+    }
+  }
+  return tmp;
```

Comment:
```suggestion
  auto tmps = RegSet::of(r0, r1, r2) - RegSet::of(src, dest);
  return tmps.first();
```
Use the force, Luke Skywalker.

Reply:
A-and ... this version actually works:

    auto tmps = RegSet::of(r0, r1, r2) - RegSet::of(src, dst);
    return *tmps.begin();

(`first()` is private :-)